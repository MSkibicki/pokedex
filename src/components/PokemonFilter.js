import React from "react";

const PokemonFilter = ({ handleInput }) => {
  return (
    <div>
      <label htmlFor="pokemon-filter">Filter pokemon by name.</label>
      <input
        type="text"
        name="filter"
        id="filter"
        placeholder="Start typing pokemon name."
        onChange={handleInput}
      />
    </div>
  );
};

export default PokemonFilter;
