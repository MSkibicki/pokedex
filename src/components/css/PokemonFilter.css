.filter {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  margin-top: 3em;
  margin-bottom: 1em;
}

.filter .input {
  height: 36px;
  width: 294px;
  background-color: #f2f7eb;
  border-radius: 10px;
  padding: 0 0 0 5px;
  margin-left: 1em;
}

.filter .input:focus {
  -webkit-box-shadow: 5px 5px 5px black;
          box-shadow: 5px 5px 5px black;
  outline: 0;
}

.filter label {
  height: 40px;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  margin-right: 1em;
  color: #f2f7eb;
  font-size: 20px;
  text-shadow: 2px 2px 2px black;
}

@media (max-width: 550px) {
  .filter {
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
        -ms-flex-direction: column;
            flex-direction: column;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
  }
}

@media (max-width: 550px) {
  label {
    margin-bottom: 1em;
    margin-right: 0;
  }
}

@media (max-width: 550px) {
  .input {
    margin-left: 0;
  }
}
/*# sourceMappingURL=PokemonFilter.css.map */