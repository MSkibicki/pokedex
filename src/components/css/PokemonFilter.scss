@import "./Global.scss";

.filter {
  display: flex;
  justify-content: center;
  margin-top: 3em;
  margin-bottom: 1em;

  .input {
    height: 36px;
    width: 294px;
    background-color: $main-color;
    border-radius: 10px;
    padding: 0 0 0 5px;
    margin-left: 1em;
  }

  .input:focus {
    box-shadow: 5px 5px 5px $black;
    outline: 0;
  }

  label {
    height: 40px;
    align-items: center;
    display: flex;
    margin-right: 1em;
    color: $main-color;
    font-size: 20px;
    text-shadow: 2px 2px 2px $black;
  }
}

@media (max-width: 550px) {
  .filter {
    flex-direction: column;
    align-items: center;
  }
}

@media (max-width: 550px) {
  .filter {
    label {
      margin-bottom: 1em;
      margin-right: 0;
    }
  }
}

@media (max-width: 550px) {
  .filter {
    .input {
      margin-left: 0;
    }
  }
}

@media (max-width: 350px) {
  .filter {
    .input {
      margin: 0 auto;
      width: 80%;
    }
  }
}

@media (max-width: 320px) {
  .filter {
    .input {
      margin: 0 auto;
      width: 80%;
    }
  }
}
