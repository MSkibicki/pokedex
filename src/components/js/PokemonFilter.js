import React from "react";
import "../css/PokemonFilter.scss";
import PropTypes from "prop-types";

const PokemonFilter = ({ handleInput }) => {
  return (
    <div className="filter">
      <label htmlFor="pokemon-filter">Filter pokemon by name.</label>
      <input
        type="text"
        name="filter"
        id="filter"
        placeholder="Start typing pokemon name."
        onChange={handleInput}
        className="input"
      />
    </div>
  );
};

PokemonFilter.propTypes = {
  handleInput: PropTypes.func,
};

export default PokemonFilter;
