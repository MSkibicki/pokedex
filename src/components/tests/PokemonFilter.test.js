import PokemonFilter from "./../js/PokemonFilter";

describe("<PokemonFilter />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<PokemonFilter handleInput={() => {}} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("calls handleInput on change", () => {
    const spy = sinon.spy();
    const wrapper = mount(<PokemonFilter handleInput={spy} />);

    wrapper.find("#filter").simulate("change");

    expect(spy.calledOnce).toBe(true);
  });

  it("displays label text correctly", () => {
    const wrapper = shallow(<PokemonFilter handleInput={() => {}} />);

    const text = wrapper.find("label").text();

    expect(text).toEqual("Filter pokemon by name.");
  });

  it("has correct div classname", () => {
    const wrapper = shallow(<PokemonFilter handleInput={() => {}} />);

    const divClass = wrapper.find("div").hasClass("filter");

    expect(divClass).toBe(true);
  });

  it("has correct input classname", () => {
    const wrapper = shallow(<PokemonFilter handleInput={() => {}} />);

    const inputClass = wrapper.find("input").hasClass("input");

    expect(inputClass).toBe(true);
  });
});
