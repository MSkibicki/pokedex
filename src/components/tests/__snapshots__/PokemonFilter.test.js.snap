// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonFilter /> renders correctly 1`] = `
<div
  className="filter"
>
  <label
    htmlFor="pokemon-filter"
  >
    Filter pokemon by name.
  </label>
  <input
    className="input"
    id="filter"
    name="filter"
    onChange={[Function]}
    placeholder="Start typing pokemon name."
    type="text"
  />
</div>
`;
